packages/modules/imodel-browser/src/containers/iModelTiles/IModelTile.tsx

```diff
@@ -85,6 +104,45 @@ export const IModelTile = ({
         }
       : undefined;
 
+  const handleClickAndAddToRecents = React.useCallback(
+    async (e: React.MouseEvent<HTMLElement>) => {
+      if (favoritesContext) {
+        try {
+          if (!accessToken) {
+            return;
+          }
+
+          const url = `${_getAPIServer(
+            apiOverrides?.serverEnvironmentPrefix
+          )}/imodels/recents/${encodeURIComponent(iModel.id)}`;
+
+          void fetch(url, {
+            method: "POST",
+            headers: {
+              authorization:
+                typeof accessToken === "function"
+                  ? await accessToken()
+                  : accessToken,
+              Accept: "application/vnd.bentley.itwin-platform.v2+json",
+            },
+          });
+        } catch (e) {
+          // swallow errors to avoid disrupting the UI
+          console.error("Failed to add iModel to recents", e);
+        }
+      }
+      onClick?.(e) ?? onThumbnailClick?.(iModel);
```

Comment:
This is a bit confusing. Why are we calling onClick and onThumbnailClick? What does onClick return?